proto/src/determined/resourcepool/v1/resourcepool.proto

```diff
@@ -0,0 +1,182 @@
+  // by the master_url option is an IP address or is not contained in the certificate
+  string master_cert_name=22;
+  // The startup script for the agent. This runs on the node the agent runs on.
+  string startup_script=23;
+  // The startup script for the agent's container. This runs in the container determined-agent runs in.
+  string container_startup_script=24;
+  // The Docker network to use for the agent when using dynamic agents.
+  string agent_docker_network=25;
+  // The docker runtime to use for the agent when using dynamic agents
+  string agent_docker_runtime=26;
+  // The docker image to use for the agent when using dynamic agents
+  string agent_docker_image=27;
+  // The Fluent docker image to use
+  string agent_fluent_image=28;
+  // The maximum idle period of agents. The master waits for this period of time
+  // before shutting down idle agents.
+  string max_idle_agent_period=29;
+  // The maximum starting period of agents. The master waits for this period of time for starting
+  // agents before retrying.
+  string max_agent_starting_period=30;
+
+  // GCP, AWS and Priority Scheduler details
+  determined.resourcepool.v1.ResourcePoolDetail details=31;
+}
+
+// Detailed information about the resource pool
+message ResourcePoolDetail {
+  // AWS-specific details
+  determined.resourcepool.v1.ResourcePoolAwsDetail aws=1;
+  // GCP-specific details
+  determined.resourcepool.v1.ResourcePoolGcpDetail gcp=2;
+  // Priority scheduler-specific details
+  determined.resourcepool.v1.ResourcePoolPrioritySchedulerDetail priority_scheduler=3;
+}
+
+// List of arbitrary user-defined tags that are added to the Determined agent instances
+message AwsCustomTag {
+  // The key of the custom tag
+  string key=1;
+  // The value of the custom tag
+  string value=2;
+}
+
+// AWS-specific details about the resource pool
+message ResourcePoolAwsDetail {
+  // The region the resource pool exists in
+  string region=1;
+  //  Size of the root volume of the Determined agent in GB
+  int32 root_volume_size=2;
+  // The AMI ID of the Determined agent
+  string image_id=3;
+  // Key for tagging the Determined agent instances
+  string tag_key=4;
+  // Value for tagging the Determined agent instances
+  string tag_value=5;
+  // Name to set for the Determined agent instances
+  string instance_name=6;
+  // The name of the SSH key registered with AWS for SSH key access to the agent instances
+  string ssh_key_name=7;
+  // Whether to use public IP addresses for the Determined agent
+  bool public_ip=8;
+  // The ID of the subnet to run the Determined agents in
+  string subnet_id=9;
+  // The ID of the security group to run the Determined agents as
+  string security_group_id=10;
+  // The Amazon Resource Name (ARN) of the IAM instance profile to attach to the agent instances.
+  string iam_instance_profile_arn=11;
+  // AWS instance type to use for dynamic agents
+  string instance_type=12;
```

Comment:
ditto enum

Reply:
I'm going to leave this one as a string. There's a lot of options and AFAICT you can't create an enum where the value of the fields should have periods in them.